x-pack/metricbeat/module/sql/query/query.go

```diff
@@ -345,19 +345,28 @@ func inferTypeFromMetrics(ms mapstr.M) mapstr.M {
 
 	for k, v := range ms {
 		switch v.(type) {
-		case float64:
+		case int, int8, int16, int32, int64, uint, uint8, uint16, uint32, uint64, float32, float64:
```

Comment:
This is where based on types, key-vals are assigned under `numeric`, `string` or `bool`